### ISSUE TITLE
400 Login Error

### ISSUE THREAD

**Issue by ghost** [hoschiCZ/instastories-backup#9]
After parsing the json I get this error. It does manage to rewrite usernames into ids.

`ERROR: got 400 when fetching stories entries! Response: {"message": "login_required", "logout_reason": 2, "status": "fail"}`

I entered the correct username and password but no cookie, since its (probably?) not needed because I don't use 2FA.

**Comment by ghost** [comment]
This is the json I use.

`{
    "filesdir": "files",
    "dbfilename": "stories.sqlite3",
    "username": "xxxx",
    "password": "xxxx",
    "quiet": false,
    "ids": {
        "xxxx": "xxx",
        "xxxx": "xxx",
        "xxx": "xxx",
        "xxxx: "xxxx",
        "xx": "xxxx",
    },
    "x-ig-capabilities": "36oD"
}
`

Looked a bit more into x-ig-capabilities. Since the prefs.json got rewritten I think I don't need it anymore. Not using it doesn't help either.

**Comment by tomglynch** [comment]
Replicated the issue. Unable to fix.

**Comment by ondrejsojka** [comment]
Looks like IG changed the API for logging in over ajax. Will look into it.

**Comment by tomglynch** [comment]
Thanks @tensojka

This may be relevant: https://stackoverflow.com/questions/49788905/what-is-the-new-instagram-json-endpoint

**Comment by lordzuqi** [comment]
is there still no fix yet ?
I'm having this error as well.

**Comment by ondrejsojka** [comment]
@lordzuqi no, still no fix yet. They totally removed the ajax login API and I haven't yet figured out what they replaced it with.

**Comment by aboutte** [comment]
Does the IG change impact both username/password and cookie authentication? I have been using cookie authentication for last several months without issue and suddenly last night I am no longer able to authenticate.

**Comment by AndCycle** [comment]
@aboutte IG just blocked any automation access that required login, now it's hopeless
